esp/bindings/http/platform/httpservice.cpp

```diff
@@ -481,6 +481,17 @@ int CEspHttpServer::processRequest()
         if ((authState == authTaskDone) || (authState == authFailed))
             return 0;
 
+        // Set Content-Security-Policy header here to prevent frame injection
+        // before any of the handler functions are called that could return a
+        // page with a parameterized frame, such as legacy ECLWatch pages
+        // returned as files. Restrict src to the current host with a wildcard
+        // port to allow ws_ecl to be embedded in a frame when it is bound to
+        // a different port.
+        StringBuffer host;
+        m_request->getHost(host);
```

Comment:
@asselitx will this be the url that the browser used to resolve the service, or could it be a local ip address?

I what to check what happens if there are multiple esps behind a load-balancer, if one request goes to one esp, and the request for the contents of the frame goes to another.

Reply:
I believe it is the browser's URL, but I will confirm and test a situation like you've outlined.

Reply:
@ghalliday , I spoke with @GordonSmith today and he pointed out that this will be correct even for ESPs behind a load balancer. The security is enforced by the browser itself, and the ESP is simply setting the CSP Header in response to the Host reported by the browser itself, which will be the same host it uses to request both the main page and the ws_ecl frame.

If this is acceptable I'll rebase to 9.2 otherwise continue with a LB test.